vhd/publishing/new-sku-and-add-image-version.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash -e
+
+required_env_vars=(
+    "SKU_PREFIX"
+    "SKU_TEMPLATE_FILE"
+    "AZURE_TENANT_ID"
+    "AZURE_CLIENT_ID"
+    "AZURE_CLIENT_SECRET"
+    "PUBLISHER"
+    "OFFER"
+    "CONTAINER_RUNTIME"
+)
+
+for v in "${required_env_vars[@]}"
+do
+    if [ -z "${!v}" ]; then
+        echo "$v was not set!"
+        exit 1
+    fi
+done
+
+if [ ! -f "$SKU_TEMPLATE_FILE" ]; then
+    echo "Could not find sku template file: ${SKU_TEMPLATE_FILE}!"
+    exit 1
+fi
+
+(set -x; ls -lhR artifacts )
+
+VHD_INFO="artifacts/vhd/windows-vhd-publishing-info.json"
+
+if [ ! -f "$VHD_INFO" ]; then
+    echo "Could not find sku template file: ${VHD_INFO}!"
+    exit 1
+fi
+
+short_date=$(date +"%y%m")
+pretty_date=$(date +"%b %Y")
+
+sku_id="${SKU_PREFIX}-${short_date}"
+
+echo "Checking if offer contains SKU: $sku_id"
+# Check if SKU already exists in offer
+(set -x; hack/tools/bin/pub skus list -p microsoft-aks -o aks-windows | jq ".[] | .planId" | tr -d '"' | tee skus.txt)
```

Comment:
We may use publisher and offer as variables, as we are receiving them as required variables?

Reply:
Let me update the PR so they we can take these in as environment variables.

Reply:
Updated!